scipy/signal/filter_design.py

```diff
@@ -353,6 +358,117 @@ def group_delay(system, w=None, whole=False):
     return w, gd
 
 
+def _validate_sos(sos):
+    """Helper to validate a SOS input"""
+    sos = np.atleast_2d(sos)
+    if sos.ndim != 2:
+        raise ValueError('sos array must be 2D')
+    n_sections, m = sos.shape
+    if m != 6:
+        raise ValueError('sos array must be shape (n_sections, 6)')
+    if not (sos[:, 3] == 1).all():
+        raise ValueError('sos[:, 3] should be all ones')
```

Comment:
We should be guaranteed this (gains should be elsewhere) from the SciPy functions even though user-defined ones might not have it. But if we're going to change that we need to do it elsewhere and discuss it since this is in 0.18 and all I did was move it.